packages/salesforcedx-vscode-core/src/commands/forceSourcePush.ts

```diff
@@ -33,8 +34,11 @@ export class ForceSourcePushExecutor extends BaseDeployExecutor {
       .withArg('force:source:push')
       .withJson()
       .withLogName('force_source_push_default_scratch_org');
-    if (this.flag === '--forceoverwrite') {
-      builder.withArg(this.flag);
+    if (
+      sfdxCoreSettings.getForcePushAndPullEnabled() ||
+      this.flag === '--forceoverwrite'
+    ) {
+      builder.withArg('--forceoverwrite');
```

Comment:
Might want to enable the force push here instead:

https://github.com/forcedotcom/salesforcedx-vscode/blob/06e7e5fa1b8ef4c19b2057d56d679003a3ccbcba/packages/salesforcedx-vscode-core/src/settings/pushOrDeployOnSave.ts#L84

and have a check like

```typescript
let command = 'sfdx.force.source.push';
if (sfdxCoreSettings.getForcePushOnSave()) {
  command += '.force'
}
vscode.commands.executeCommand(command);
```

Reply:
done